webapp/utils/channel_utils.jsx

```diff
@@ -190,40 +190,44 @@ export function showManagementOptions(channel, isAdmin, isSystemAdmin, isChannel
     return true;
 }
 
-export function showDeleteOption(channel, isAdmin, isSystemAdmin, isChannelAdmin, userCount) {
+export function showDeleteOptionForCurrentUser(channel) {
+    const isTeamAdmin = TeamStore.isTeamAdminForCurrentTeam();
+    const isSystemAdmin = UserStore.isSystemAdminForCurrentUser();
+    const isChannelAdmin = ChannelStore.isChannelAdminForCurrentChannel();
+
```

Comment:
@grundleborg fixed in https://github.com/mattermost/platform/pull/7050/commits/17abcc9a424c800601b14eb69349026af90b62c1